server/onlineUsers.js

```diff
@@ -1,2 +1,42 @@
-const onlineUsers = [];
+class OnlineUsers {
+  // Stores a dictionary of structure:
+  // {
+  //   userId :
+  //   {
+  //     socketId: int
+  //   }
+  // }
+  constructor() {
+    this.users = {};
+  }
+
+  // checks if userId exists
+  includes(userId) {
+    return userId in this.users;
+  }
+
+  // returns true if added, false if not added
+  addUser(userId, socketId) {
+    if (this.includes(userId)) return false;
```

Comment:
This may prevent the same user to be logged into multiple devices simultaneously.